src/ports/postgres/modules/deep_learning/test/madlib_keras_model_selection.sql_in

```diff
@@ -580,6 +741,44 @@ FROM (SELECT count(*) cnt FROM iris_multiple_model_info
 WHERE compile_params = $MAD$loss='categorical_crossentropy', optimizer='Adam(lr=0.01)', metrics=['accuracy']$MAD$::text
 AND fit_params = $MAD$batch_size=32, epochs=1$MAD$::text) info;
 
+-- Test with caching when number of configs(4) larger than number of segments(3)
+DROP TABLE if exists iris_multiple_model, iris_multiple_model_summary, iris_multiple_model_info;
+SELECT madlib_keras_fit_multiple_model(
+	'iris_data_packed',
+	'iris_multiple_model',
+	'mst_table_4row',
+	3,
+	FALSE, NULL, NULL, NULL, NULL, NULL,
+	TRUE
+);
```

Comment:
Yes, this will still pass as it asserts on the number of models trained. Eventhough  for a cluster size > 3 segments, it might not test the use case intended in this specific test but it would not fail.